rcl/test/rcl/test_subscription.cpp

```diff
@@ -565,28 +765,54 @@ TEST_F(CLASSNAME(TestSubscriptionFixture, RMW_IMPLEMENTATION), test_subscription
   }
   ASSERT_TRUE(wait_for_subscription_to_be_ready(&subscription, context_ptr, 10, 100));
   {
-    test_msgs__msg__Strings msg;
-    test_msgs__msg__Strings * msg_loaned;
-    test_msgs__msg__Strings__init(&msg);
-    OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
-    {
-      test_msgs__msg__Strings__fini(&msg);
-    });
-
+    auto patch_take =
+      mocking_utils::prepare_patch("lib:rcl", rmw_take_loaned_message_with_info);
+    auto patch_return =
+      mocking_utils::prepare_patch("lib:rcl", rmw_return_loaned_message_from_subscription);
     // Only if rmw supports the functionality
-    if (rcl_subscription_can_loan_messages(&subscription)) {
-      ret = rcl_take_loaned_message(
-        &subscription, reinterpret_cast<void **>(&msg_loaned), nullptr, nullptr);
-      ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
-      ASSERT_EQ(
-        std::string(test_string),
-        std::string(msg_loaned->string_value.data, msg_loaned->string_value.size));
-    } else {
-      ret = rcl_take(&subscription, &msg, nullptr, nullptr);
-      ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
-      ASSERT_EQ(
-        std::string(test_string), std::string(msg.string_value.data, msg.string_value.size));
+    if (!rcl_subscription_can_loan_messages(&subscription)) {
```

Comment:
Do you think it's a good idea to mock the loaned functions this way? might be a bit misleading. I think it would be helpful adding a comment about the purpose of this block of code for future readers.

Reply:
Well, we need a way to test the happy path when using loaned messages for subscription at the `rcl` level. By mocking like this, we eliminate code duplication and the test remains true i.e. it tests what it is supposed to test. Loaning APIs say nothing about how the middleware performs such allocations, so I think we're in the clear.

I can add a comment for sure though.

Reply:
Got it, I agree. Please remember adding the comment with this before merging.

Reply:
Yes! See 2f3875f.